doc/source/numpy_2_0_migration_guide.rst

```diff
@@ -83,6 +83,48 @@ used to explicitly implement different behavior on NumPy 1.x and 2.0.
 
 Please let us know if you require additional workarounds here.
 
+.. _migration_c_descr:
+
+The ``PyArray_Descr`` struct has been changed
+---------------------------------------------
+One of the most impactful C-API changes is that the ``PyArray_Descr`` struct
+is no more opaque to allow us to add additional flags and have
```

Comment:
Should this say it is now opaque instead of no longer opaque? Unless I am misinterpreting the word opaque, but I would assume you want to make it more opaque to allow changing it later without breaking user code again

Reply:
Thanks, it was meant to be a "now", not a "no".  It is mostly changed really, but I think it makes sense to write it as opaque.
(I.e. `elsize` is available if you set `-DNPY_TARGET_VERSION=NPY_2_0_API_VERSION`, but that should be relevant only for authors of new-style DTypes.)